src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -151,8 +165,12 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
 
       // Non-static properties.
       for (propertySpec in propertySpecs) {
+        if (primaryConstructor?.parameters?.find { it.name == propertySpec.name && it.type == propertySpec.type } != null) {
```

Comment:
I wonder if would simplify to index inlined parameters? Like build a `Map<String, PropertySpec>` ?